app/User.php

```diff
@@ -32,4 +34,33 @@ public function roles()
     {
         return $this->belongsToMany(Role::class);
     }
+
+    public function tokens()
+    {
+        return $this->hasMany(Token::class, 'user_id');
+    }
+
+    public static function getUserIdFromToken(Request $request)
```

Comment:
this function should be a part of authentication middleware

Reply:
I do not know how to get a middleware feature so I can call it from the model.